codes/c/chapter_backtracking/n_queens.c

```diff
@@ -0,0 +1,92 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+
+#define MAX_N 20
+
+/* 棋盘结构 */
+typedef struct {
+    char state[MAX_N][MAX_N][2];
+    int n;
+    bool cols[MAX_N];         //记录是否有皇后
+    bool diags1[2*MAX_N - 1]; //记录主对角线是否有皇后
+    bool diags2[2*MAX_N - 1]; //记录副对角线是否有皇后
+} Board;
+
+typedef struct {
+    char ****res;
+    int count;
+} Results;
+
+/* 回溯算法：N 皇后 */
+void backtrack(int row, Board *board, Results *results) {
```

Comment:
Refer to other language code implementations, such as [cpp](https://github.com/krahets/hello-algo/blob/main/codes/cpp/chapter_backtracking/n_queens.cpp)